main.tf

```diff
@@ -262,136 +231,27 @@ resource "kubernetes_cluster_role_binding" "user" {
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-# GENERATE TLS CERTIFICATES FOR USE WITH TILLER
-# This will use kubergrunt to generate TLS certificates, and upload them as Kubernetes Secrets that can then be used by
-# Tiller.
-# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-resource "null_resource" "tiller_tls_certs" {
-  provisioner "local-exec" {
-    command = <<-EOF
-      kubergrunt tls gen --ca --namespace kube-system --secret-name ${local.tls_ca_secret_name} --secret-label gruntwork.io/tiller-namespace=${local.tiller_namespace} --secret-label gruntwork.io/tiller-credentials=true --secret-label gruntwork.io/tiller-credentials-type=ca --tls-subject-json '${jsonencode(var.tls_subject)}' ${local.tls_algorithm_config} ${local.kubectl_auth_config}
-
-      kubergrunt tls gen --namespace ${local.tiller_namespace} --ca-secret-name ${local.tls_ca_secret_name} --ca-namespace kube-system --secret-name ${local.tls_secret_name} --secret-label gruntwork.io/tiller-namespace=${local.tiller_namespace} --secret-label gruntwork.io/tiller-credentials=true --secret-label gruntwork.io/tiller-credentials-type=server --tls-subject-json '${jsonencode(var.tls_subject)}' ${local.tls_algorithm_config} ${local.kubectl_auth_config}
-    EOF
-
-    # Use environment variables for Kubernetes credentials to avoid leaking into the logs
-    environment = {
-      KUBECTL_SERVER_ENDPOINT = data.template_file.gke_host_endpoint.rendered
-      KUBECTL_CA_DATA         = base64encode(data.template_file.cluster_ca_certificate.rendered)
-      KUBECTL_TOKEN           = data.template_file.access_token.rendered
-    }
-  }
-}
-
 # ---------------------------------------------------------------------------------------------------------------------
-# DEPLOY TILLER TO THE GKE CLUSTER
+# DEPLOY A SAMPLE CHART
 # ---------------------------------------------------------------------------------------------------------------------
 
-module "tiller" {
-  source = "github.com/gruntwork-io/terraform-kubernetes-helm.git//modules/k8s-tiller?ref=v0.5.0"
-
-  tiller_tls_gen_method                    = "none"
-  tiller_service_account_name              = kubernetes_service_account.tiller.metadata[0].name
-  tiller_service_account_token_secret_name = kubernetes_service_account.tiller.default_secret_name
-  tiller_tls_secret_name                   = local.tls_secret_name
-  namespace                                = local.tiller_namespace
-  tiller_image_version                     = local.tiller_version
-
-  # Kubergrunt will store the private key under the key "tls.pem" in the corresponding Secret resource, which will be
-  # accessed as a file when mounted into the container.
-  tiller_tls_key_file_name = "tls.pem"
-
-  dependencies = [null_resource.tiller_tls_certs.id, kubernetes_cluster_role_binding.user.id]
+data "helm_repository" "bitnami" {
```

Comment:
NIT: maybe worth a comment about what is happening here - would probably be helpful for folks new to Helm concepts.

Reply:
Specifically that its going to look up a chart from a chart repository.

Reply:
NIT: There is actually a critical issue with this data source (https://github.com/terraform-providers/terraform-provider-helm/issues/438 and https://github.com/terraform-providers/terraform-provider-helm/issues/416), and they are actually considering removing it entirely. I've found that it is much more robust and stable to directly use the repo URL in the `repository` attribute of the `helm_release` resource instead.

Reply:
Thanks @yorinasub17 for pointing this out. I'll fix this!